src/ColorVectorSpace.jl

```diff
@@ -237,6 +238,112 @@ an array: `complement.(x)`
+ColorTypes.base_colorant_type(::Type{<: Complement{C, <: Any, N}}) where {N,C} = Complement{base_colorant_type(C), <: Any, N}
+
+Base.oneunit(::Type{Complement{C,T,N}}) where {C,T,N} = Complement{C,T,N}(zero(C))
+Base.zero(::Type{Complement{C,T,N}})  where {C,T,N} = Complement{C,T,N}(oneunit(C))
+Base.one(::Type{Complement{C,T,N}})  where {C,T,N} = Complement{C,T,N}(zero(C))
+ColorTypes.nan(::Type{Complement{C,T,N}})  where {C,T,N} = Complement{C,T,N}(nan(C))
+
+complement(c::Complement) = parent(c)
+Base.convert(::Type{C}, c::Complement) where {C <: Colorant} = convert(C, complement(parent(c)))
+Base.convert(::Type{Complement}, c::C) where {T ,N, C <: Complement{<: Colorant,T,N}} = c
+Base.convert(::Type{Complement}, c::C) where {T, N, C <: Colorant{T,N}} = Complement(complement(c))
+Base.convert(::Type{Complement{C}}, c::C) where {T, N, C <: Colorant{T,N}} = Complement(complement(c))
+Base.convert(::Type{Complement{C,T}}, c::C) where {T, N, C <: Colorant{T,N}} = Complement(complement(c))
+Base.convert(::Type{Complement{C,T,N}}, c::C) where {T, N, C <: Colorant{T,N}} = Complement(complement(c))
+
+"""
+    reinterpret(::Type{<: Complement}, array::AbstractArray{<: Colorant{T,N}})
+
+Reinterpret an `AbstractArray`` of `Colorant{T,N}`, C, as an `AbstractArray`` of `Complement{C,T,N}`.
+
+For a `<: Colorant{T,N}`, the first argument could be one of
+* `Complement`,
+* `Complement{C}`,
+* `Complement{C, T}`
+* `Complement{C, T, N}` (Base implementation of `reinterpret`)
+"""
+Base.reinterpret(::Type{Complement}, array::AbstractArray{C}) where {T, N, C <: Colorant{T,N}} =
+    reinterpret(Complement{C,T,N}, array)
+# Address ambiguity
+Base.reinterpret(::Type{Complement}, array::Base.ReinterpretArray{C, N, S, A, false} where {N, S, A<:AbstractArray{S, N}}) where {TT, NN, C<:ColorTypes.Colorant{TT, NN}} =
+    reinterpret(Complement{C,TT,NN}, array)
+
+"""
+    ComplementArray{T,N,A} <: AbstractArray{T,N}
+    ComplementArray(::AbstractArray{<: Complement})
+
+ComplementArray is an `AbstractArray{T}` that wraps an `AbstractArray` with a
+`Complement{T}` element type. This allows for a `AbstractArray{Complement{T}}`
+to be wrapped into an `AbstractArray{T}` without additional allocations.
+"""
+struct ComplementArray{T,N,A} <: AbstractArray{T,N}
+    parent::A
+    ComplementArray(parent::A) where {T,N,A <: AbstractArray{<:Complement{T},N}} = new{T,N,A}(parent)
+end
+Base.parent(a::ComplementArray) = a.parent
+Base.size(a::ComplementArray) = size(a.parent)
+function Base.getindex(a::ComplementArray{T}, I::Vararg{Int,N})::T where {T,N}
+    getindex(a.parent, I...)
+end
+function Base.setindex!(a::ComplementArray{T}, v, I::Vararg{Int, N}) where {T,N}
+    setindex!(a.parent, convert(eltype(a.parent), v))
+end
+Base.IndexStyle(a::ComplementArray) = IndexStyle(a.parent)
+
+function Base.show(io::IO, mime::MIME"image/svg+xml", c::Complement{C}) where C
+    show(io, mime, C(c))
```

Comment:
What happens without this method? I would think that if you have all the abstractions correct, this Just Works without needing a new method. Similar thoughts about the other methods below.